app/src/main/java/org/akvo/flow/activity/SignatureActivity.java

```diff
@@ -56,26 +64,59 @@ public void onClick(View v) {
                 finish();
             }
         });
-        findViewById(R.id.save).setOnClickListener(new View.OnClickListener() {
+        sendButton = findViewById(R.id.save);
+        sendButton.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                if (!mSignatureView.isEmpty()) {
+                if (!mSignatureView.isEmpty() && !TextUtils
+                        .isEmpty(nameEditText.getText().toString())) {
                     save();
                 }
             }
         });
+        nameEditText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count,
+                    int after) {
+                //EMPTY
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                //EMPTY
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                updateSendButton();
+            }
+        });
+        mSignatureView.setListener(this);
+    }
+
+    private void updateSendButton() {
+        boolean isNameEmpty = TextUtils.isEmpty(nameEditText.getText().toString());
+        boolean signatureEmpty = mSignatureView.isEmpty();
+        sendButton.setEnabled(!isNameEmpty && !signatureEmpty);
     }
 
     private void clear() {
         mSignatureView.clear();
+        updateSendButton();
```

Comment:
That's a good point, but this is the button to clear the drawing, I am inclined to leave it like that since you can edit the text some other way, here is a screenshot, what do you think?
![screenshot_1497533360](https://user-images.githubusercontent.com/923280/27183617-86c3b57c-51df-11e7-9b2c-b9ccf114ea79.png)